pkg/ipcache/cidr.go

```diff
@@ -61,8 +62,12 @@ func allocateCIDRs(prefixes []*net.IPNet) ([]*identity.Identity, error) {
 		}
 
 		prefixStr := prefix.String()
+
 		// Figure out if this call needs to be able to update the selector cache synchronously.
-		id, isNew, err := cache.AllocateIdentity(context.Background(), nil, cidr.GetCIDRLabels(prefix))
+		allocateCtx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
For this case, the timeout is not (expected to be) related to the kvstore so in this package it could potentially have its own `IPAllocationTimeout`.